pkg/stanza/operator/parser/regex/cache.go

```diff
@@ -154,14 +161,21 @@ var _ limiter = &atomicLimiter{count: &atomic.Uint64{}}
 func (l *atomicLimiter) init() {
 	// start the reset go routine once
 	l.start.Do(func() {
+		l.done = make(chan struct{})
```

Comment:
nit: I guess I would have expected this to happen within `newStartedAtomicLimiter`

Reply:
Good call. I had it there initially but moved it because of one test with a strange setup. I've moved it back and handled the test differently.